File: src/main/java/com/EmosewaPixel/pixellib/recipes/EnergyMachineRecipe.java
package com.EmosewaPixel.pixellib.recipes;

//Energy Machine Recipes are Machine Recipes that take power
public class EnergyMachineRecipe extends SimpleMachineRecipe {
    private int energyPerTick;
    public static EnergyMachineRecipe EMPTY = new EnergyMachineRecipe(new Object[0], new Float[0], new Object[0], new Float[0], 0, 0);

    public EnergyMachineRecipe(Object[] input, Float[] consumeChances, Object[] output, Float[] outputChances, int time, int energyperTick) {
        super(input, consumeChances, output, outputChances, time);
        this.energyPerTick = energyperTick;
    }

    @Override
    public boolean isEmpty() {
        return this == EMPTY;
    }

    public int getEnergyPerTick() {
        return energyPerTick;
    }

    public void setEnergyPerTick(int energyPerTick) {
        this.energyPerTick = energyPerTick;
    }
}

File: src/main/java/com/EmosewaPixel/pixellib/recipes/EnergyMachineRecipe.kt
package com.EmosewaPixel.pixellib.recipes

//Energy Machine Recipes are Machine Recipes that take power
class EnergyMachineRecipe(input: Array<Any>, consumeChances: Array<Float>, output: Array<Any>, outputChances: Array<Float>, time: Int, var energyPerTick: Int) : SimpleMachineRecipe(input, consumeChances, output, outputChances, time) {
    override val isEmpty = this === EMPTY

    companion object {
        @JvmField
        var EMPTY = EnergyMachineRecipe(arrayOf(), arrayOf(), arrayOf(), arrayOf(), 0, 0)
    }
}